### PR TITLE
[Feature] Special note for processes

### LINKED ISSUE #7986: ✨ Special note for processes
## ✨ Feature

Admin portal and Job advertisement changes to add a special note. 

New section is called "Special note for this process"

## 🎨 [Figma file for candidate side](https://www.figma.com/file/CdYmWlP6SpT17bw7Q9411Z/Candidates---job-poster?type=design&node-id=122-5501&mode=design&t=DFocFq8puY6U6iXs-4)

# 👩🏻‍💻Admin portal 

## Special note for this process
Heading: Special note for this process
Body copy: Most job advertisements will not require a special note. This section is for special circumstances only. Examples of this special note include identifying if a process may be used to hire in multiple classifications or if the process is limited to application from a specific equity group. 

Confirmation box: Not required 
confirmation box input label: Confirmation of need for special note
copy: I need a special note. I have reviewed my job advertisement preview and I confirm that this information is not covered anywhere else on the job advertisement. 

### 📸 Admin portal - edit pool 
false / empty
Screenshot 2023-09-21 at 11 28 23 AM

true
Screenshot 2023-09-21 at 11 29 56 AM

# 💼Job advertisement 

## Special note for this process

Component: Alert 
Task: Info 
Heading: Special note for this process
Body copy is filled out by admin. If nothing is submitted this section will be hidden. 

**Please note:** The side nav bar should also display this section

### 📸 Candidate - Job advertisement 
Screenshot 2023-09-21 at 11 30 45 AM
Screenshot 2023-09-21 at 11 32 13 AM

### Acceptance Criteria
- [ ] Admin pool creation page has new OPTIONAL section to add English and French copy for the new "special note for this process" hidden behind check box.
- [ ] New section on job poster "Special note for this process"
- [ ] Navigation sidebar shows new "Special note for this process" section
- [ ] New section AND sidebar (for new section) are hidden if this section was left blank on the admin side for pool creation.
- [ ] Review from designer - @brookecameron
- [ ] input `special_note_for_this_process` (en, fr) added to pool (not required, and not required to publish)
- [ ] migration to update model and schema
- [ ] update pool factory

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #7986

## 👋 Introduction

- Adds `special_note` localized string ["en" => "", "fr" => ""] to Pool model (migration, model, factory).
- Adds new sections to admin side on edit pool (draft and published), and view pool.
- Adds new section to applicant side on pool advertisement page (below opportunity section).

## 🧪 Testing

1. Login as admin
2. On admin side, create or find a draft pool. Ensure new special note section appears correctly. The textboxes should only appear if the confirmation box is selected. If the pool is published the section should be disabled (checkbox and textboxes).
3. On applicant side, find published pool and ensure section appears correctly. If the pool **has** a special note, then an alert box should appear under opportunity summary section.

## 📸 Screenshot

Admin - Edit Pool (Draft)
image

Admin - View Pool (Published)
image

Applicant - Pool Advertisement
image

### CODE REVIEW

File: apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx [hunk 1]
```diff
@@ -429,6 +447,27 @@ export const PoolPoster = ({
                 />
               </div>
             </TableOfContents.Section>
+            {showSpecialNote && (
+              <TableOfContents.Section id={sections.specialNote.id}>
+                <div>
+                  <Alert.Root
+                    type="info"
+                    data-h2-shadow="base(none)"
+                    data-h2-margin="base(0, -x1, 0, -x1)"
```
Comment (esizer) [inline apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx:456]: I think we could probably remove the margin here. Looks a little odd when it extends past the content bounds.

**Review by esizer — CHANGES_REQUESTED** [review]
Nice! This is looking great. I have one small suggestion about the alert.

Also, should we have some validation for publishing the pool when it has a special note? Right now there is nothing stopping me from entering either French or English and not providing copy for the inverse. Or is the idea that some pools could have a special note for only one language?

File: apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx (hunk 1 above)
Comment (yonikid15) [reply apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx]: Opps missed that, thanks! Fixed by [c8f964f](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8006/commits/c8f964f6e9e379116e89a87c15145faeea45a688)

**Comment by yonikid15** [comment]
> Also, should we have some validation for publishing the pool when it has a special note? Right now there is nothing stopping me from entering either French or English and not providing copy for the inverse. Or is the idea that some pools could have a special note for only one language?

Nice catch I missed that on the backend. Fixed by [89612cb](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8006/commits/89612cb0d2bbb0300f5953166cdeae377f25ce46).

**Review by esizer — APPROVED** [review]
LGTM! 👍

File: api/app/GraphQL/Validators/PoolIsCompleteValidator.php [hunk 2]
```diff
@@ -79,6 +81,8 @@ public function messages(): array
             'key_tasks.fr.required' => 'FrenchWorkTasksRequired',
             'your_impact.en.required' => 'EnglishYourImpactRequired',
             'your_impact.fr.required' => 'FrenchYourImpactRequired',
+            'special_note.en.required' => 'EnglishYourImpactRequired',
+            'special_note.fr.required' => 'FrenchYourImpactRequired',
```
Comment (vd1992) [inline api/app/GraphQL/Validators/PoolIsCompleteValidator.php:85]: Little copypaste leftover, I think for the `special_note.en` you want to use the added `EnglishSpecialNoteRequired` and the same with French.

Comment (yonikid15) [reply api/app/GraphQL/Validators/PoolIsCompleteValidator.php]: Nice catch! Fixed by [8d10095](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8006/commits/8d100950636f94721ab3e72fa481537a1808de75)

**Pull request merged**